### PR TITLE
Journalist can create article

### PR DESCRIPTION
**Pull request by erikbjoern**
# User story 
[PT Link](https://www.pivotaltracker.com/story/show/172888071)
```
As a journalist,
In order to write articles
I would like to have an article creation form  
```
# Feature
* Journalist can create articles
* With title and body, both fields required
* Displays error message from API when any field is left out
* They are posted to the main news site as they are submitted 

# Additions in this PR
* `CreateArticle` component (the page where journalist write an article)
* `WritingForm` component (placed on the `CreateArticle` page)
* `Header` component (always visible on top of site)
* Scripts for running Cypress tests properly

## Packages added
* axios
* semantic-ui-css
* semantic-ui-react
* start-server-and-test

# Screenshots
Screenshot 2020-05-24 at 11 05 24

### CODE REVIEW

**Review by kermit-klein — APPROVED** [review]

File: src/components/CreateArticle.jsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      setMessage(response.data.message);
+    } catch (error) {
+      setMessage(error.response.data.message);
+    }
+  };
+
+  return (
+    <Container className="writing-container">
+      <Form onSubmit={onSubmitHandler}>
+        <Form.Field>
```
Comment (oliverochman) [inline src/components/CreateArticle.jsx]: I need u to extract the form to a separate component

File: cypress/integration/journalistCanCreateArticle.feature.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    cy.get("#post").click();
+    cy.get("#message").should("contain", "Article successfully created!");
+  });
+
+  it("unsuccessfully without entering any title", () => {
+    cy.get("textarea#body").type(
+      "This is the body this is the body this is the body this is the body this is the body."
+    );
+    cy.get("#post").click();
+    cy.get("#message").should("not.be.visible");
```
Comment (oliverochman) [inline cypress/integration/journalistCanCreateArticle.feature.js]: You need to test for a error message here.

File: cypress/integration/journalistCanCreateArticle.feature.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  });
+
+  it("unsuccessfully without entering any body text", () => {
+    cy.get("input#title").type("This is the title");
+    cy.get("#post").click();
+    cy.get("#message").should("not.be.visible");
```
Comment (oliverochman) [inline cypress/integration/journalistCanCreateArticle.feature.js]: same here

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: cypress/integration/journalistCanCreateArticle.feature.js (hunk 2 above)
Comment (erikbjoern) [reply cypress/integration/journalistCanCreateArticle.feature.js]: The input fields have the 'required' property, so we can't send off an incomplete request. Should we change that?

File: src/components/CreateArticle.jsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import React from 'react'
+import WritingForm from './WritingForm'
+import { Container } from 'semantic-ui-react'
+
+const CreateArticle = () => {
+
```
Comment (oliverochman) [inline src/components/CreateArticle.jsx:19]: this is not what I meant by extracting it to another component.

Comment (oliverochman) [reply src/components/CreateArticle.jsx:19]: please only extract the form

**Review by oliverochman — CHANGES_REQUESTED** [review]

**Review by jysmys — APPROVED** [review]

File: src/components/CreateArticle.jsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+import WritingForm from './WritingForm'
+import { Container } from 'semantic-ui-react'
+
+const CreateArticle = () => {
+  const [message, setMessage] = useState("");
+
+  const onSubmitHandler = async (e) => {
+    try {
+      const response = await axios.post("/articles", {
+        params: { title: e.target.title.value, body: e.target.body.value },
```
Comment (ViaMarcus) [inline src/components/CreateArticle.jsx]: Unwrap the title and body, params {} wrapper does not work with backend.

**Review by ViaMarcus — CHANGES_REQUESTED** [review]

File: src/components/CreateArticle.jsx [hunk 6]
```diff
@@ -9,7 +9,7 @@ const CreateArticle = () => {
   const onSubmitHandler = async (e) => {
     try {
       const response = await axios.post("/articles", {
-        params: { title: e.target.title.value, body: e.target.body.value },
+        title: e.target.title.value, body: e.target.body.value
```
Comment (kermit-klein) [inline src/components/CreateArticle.jsx:12]: makes sense

**Review by kermit-klein — APPROVED** [review]

**Review by ViaMarcus — APPROVED** [review]

**Review by designerofthing — APPROVED** [review]
looks good

**Review by PaulineBA — APPROVED** [review]

**Review by oliverochman — APPROVED** [review]

**Pull request merged**